### PR TITLE
Added a form component and how to use it in the dummy documentation page

### PR DESCRIPTION
**Pull request by JohnRodney**
Added a form component for using materialize forms. More functionality to come soon. Demo of the form is available on the dummy routes documentation. Basicly clone the page, install ember, server the page then visit localhost:4200 and you will see the link to the forms page with a working example and documentation on how to use it. I.E. What kind of objects this component expects you to give it.

### CODE REVIEW

**Comment by DanyHunter** [comment]
I like your approach. :+1: Other than fixing a few lines of indenting it looks good.

File: app/components/materialize-form.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  didInsertElement: function(){
+    this.$('.datepicker').pickadate({
+      selectMonths: true,
+      selectYears: this.getYears(),
+    });
+    this.$('select').material_select();
+    this.getChildIds();
+  },
+  getYears: function(){
+        var tempInp = this.get('inputs');
```
Comment (DanyHunter) [inline app/components/materialize-form.js]: indentation off

File: app/components/materialize-form.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  inputs: [{input: "First Name", isText: true}, {input: "Last Name", isText: true}, {input: "Email", isText: true}, {input: "Phone Number", isText: true},
```
Comment (DanyHunter) [inline app/components/materialize-form.js]: Feel free to use more than one line to declare variables. for example:
```
inputs: [
{ input: "First Name", isText: true },
{ input: "Email, isText: true}, 
// another line,
```

**Comment by DanyHunter** [comment]
It would be nice if we could tap into the ember view layer logic to build the form instead of using jquery dom manipulation after the element rendered. One way I would go about this is to try to override the `render` method of the component. This method takes care of building the template so It's a good place to start.

**Comment by JohnRodney** [comment]
Thats a good idea I'll look into overriding the render function this weekend or a weekday afternoon if I get some time. I'll reduce our calls on the dom and maybe shorten the amount of code to use.  I'll first look into dynamically assigning an ID and then for attribute... if that solution is available then we won't even need to override the render.

**Comment by MatthewHager** [comment]
Not sure if it will help, but we should look at how ember paper handles forms: https://github.com/miguelcobain/ember-paper.

Also:
* https://github.com/dockyard/ember-easy-form
* https://github.com/indexiatech/ember-forms

**Comment by JohnRodney** [comment]
I updated the form on my master. I ended up 1. Fixing all the indentation problems (I hope). 2. Removing all calls on the dom and just using dasherize and dynamically generated IDs and For fields. 3. Added a method parameter option to sendAction back to the controller who can catch all the form data. 4. Updated the documentation page to show example of controller catching the data from the form then rendering it underneath.

Unfortunately the changes pushed to master (cause I'm a git n00b) and now I want to just delete this PR and send another from my master with the appropriate changes.

**Pull request closed without merging**